tests/integration/test_bootstrap_interface.py

```diff
@@ -0,0 +1,172 @@
+# 10 bytes received from [::1]:5685
+# 44 04 D9 D9  D9 D9 AF 5C  B1 30  D......\.0
+#  -> State: STATE_BOOTSTRAPPING
+# 10 bytes received from [::1]:5685
+# 44 04 D9 DA  DA D9 AF 5C  B1 31  D......\.1
+#  -> State: STATE_BOOTSTRAPPING
+# 122 bytes received from [::1]:5685
+# 44 03 D9 DB  DB D9 AF 5C  B1 30 01 31  11 6E FF 5B   D......\.0.1.n.[
+# 7B 22 62 6E  22 3A 22 2F  30 2F 31 2F  22 2C 22 6E   {"bn":"/0/1/","n
+# 22 3A 22 30  22 2C 22 76  73 22 3A 22  63 6F 61 70   ":"0","vs":"coap
+# 3A 2F 2F 6C  6F 63 61 6C  68 6F 73 74  3A 35 36 38   ://localhost:568
+# 33 22 7D 2C  7B 22 6E 22  3A 22 31 22  2C 22 76 62   3"},{"n":"1","vb
+# 22 3A 66 61  6C 73 65 7D  2C 7B 22 6E  22 3A 22 31   ":false},{"n":"1
+# 30 22 2C 22  76 22 3A 31  7D 2C 7B 22  6E 22 3A 22   0","v":1},{"n":"
+# 32 22 2C 22  76 22 3A 33  7D 5D                      2","v":3}]
+#  -> State: STATE_BOOTSTRAPPING
+# 103 bytes received from [::1]:5685
+# 44 03 D9 DC  DC D9 AF 5C  B1 31 01 31  11 6E FF 5B   D......\.1.1.n.[
+# 7B 22 62 6E  22 3A 22 2F  31 2F 31 2F  22 2C 22 6E   {"bn":"/1/1/","n
+# 22 3A 22 30  22 2C 22 76  22 3A 31 7D  2C 7B 22 6E   ":"0","v":1},{"n
+# 22 3A 22 31  22 2C 22 76  22 3A 33 30  30 7D 2C 7B   ":"1","v":300},{
+# 22 6E 22 3A  22 36 22 2C  22 76 62 22  3A 66 61 6C   "n":"6","vb":fal
+# 73 65 7D 2C  7B 22 6E 22  3A 22 37 22  2C 22 76 73   se},{"n":"7","vs
+# 22 3A 22 55  22 7D 5D                                ":"U"}]
+#  -> State: STATE_BOOTSTRAPPING
+# 11 bytes received from [::1]:5685
+# 44 02 D9 DD  DD D9 AF 5C  B2 62 73  D......\.bs
+#  -> State: STATE_BOOTSTRAPPING
+# Opening connection to server at localhost:5683
+#  -> State: STATE_REGISTERING
+# 11 bytes received from [::1]:5683
+# 64 5F B2 69  69 B2 AF 5C  D1 0E 0B  d_.ii..\...
+#  -> State: STATE_REGISTERING
+# 16 bytes received from [::1]:5683
+# 64 41 B2 6A  69 B2 AF 5C  82 72 64 01  30 D1 06 13   dA.ji..\.rd.0...
+#  -> State: STATE_READY
+#  -> State: STATE_READY
+#  -> State: STATE_READY
+
+@pytest.mark.skip(reason="executed twice for unknown reason. Fails second time")
+def test_client_initiated_bootstrap(lwm2mbootstrapserver, lwm2mclient_boot):
+    """LightweightM2M-1.1-int-0
+    Test the Client capability to connect the Bootstrap Server according to the
+    Client Initiated Bootstrap Mode"""
+
+    # Possible improvements:
+    # - remove (if possible) 10s sleep
+    # - update client to print "2.04 changed" to improve §A test
+    # - investigate if sending DELETE /0, /1 is correct
+    # - use and check "0-SetOfValue"
+    # - check received WRITE operations instead of sent WRITE operations
+    # - investigate if PUT /0/1 is correct or if it should be PUT /0 (same for PUT /1/1)
+    # - since CLI output needs to be checked in sequence §B and §C checks happen §B1,§C1,§B2,§C2 (same CLI).
+    # - check payload for discover response
+    # - make testcase robust to handle if blocktransfer is used (better to disable/use new default 1024 value)
+    # - check for inconsistencies in §E
```

Comment:
I feed like many of this things should be implemented before we can merge this.

Also, this should be placed in a more prominent place (#586 to the rescue!) and somehow reflected in the commit message too.

Reply:
Did you already address those issues?

Reply:
The integration tests work by parsing the output of the example binaries, this limits what is possible to do. E.g. there is no support to check actual packets. Only information that is printed can be checked etc. What is implemented is what I think is correct or as close as possible (capturing the spirit of the test).